app/main/views/suppliers.py

```diff
@@ -119,6 +119,40 @@ def view_signed_agreement(supplier_id, framework_slug):
     )
 
 
+@main.route('/suppliers/<supplier_id>/agreements/<framework_slug>/on-hold', methods=['POST'])
```

Comment:
Cool. I was missing the fact you could pull out `supplier_framework.agreementId` in the template. Makes things a lot tidier too.